py/torch_tensorrt/dynamo/conversion/impl/select.py

```diff
@@ -428,3 +429,41 @@ def scatter(
     set_layer_name(scatter_layer, target, name + "_scatter_layer", source_ir)
     out = scatter_layer.get_output(0)
     return out
+
+
+def index_put_converter(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input_tensor: TRTTensor,
+    indices: Union[TRTTensor, Tuple[TRTTensor, ...]],
+    values: TRTTensor,
+    accumulate: bool = False,
+) -> TRTTensor:
+    # Reshape indices to add an extra dimension if necessary (indices is a Tuple of ITensors)
+    reshaped_indices = []
+    for i, each_input in enumerate(indices):
```

Comment:
Since `indices` is possible to be `ITensor` per the schema, you may not be able to iterate an ITensor.
In the test case, you can try to change the line 173 to `inputs=[source_tensor, indices_tensor, value_tensor],`.
It's kind of similar to the `offsets` in the annoying `embedding_bag`. You can think about how to use native TRT Layers to do this, like ILoop.

Reply:
Besides, what blocks you when `accumulate=True`?

Reply:
When `accumulate=True`, if there are duplicate pairs of index in `indices`, the corresponding `values` should be summed and then removed from the elements. Therefore, I aimed to obtain `indices` without duplicated pairs and corresponding modified `values`, and then use these to input into the scatter layer. However, I encountered difficulties in implementing the for loop to check for duplicate pairs of index in `indices`.

Reply:
Thanks for the detailed explanations! Yes you are right, the `indices` should be list or tuple, and thus it could be iterated over. Then your current implementation LGTM.
> One more question I have is about the type definition of indices when it is a tuple of tensors. Is it correct to define indices as Union[TRTTensor, Tuple[TRTTensor, ...]]?

I think it could be `Sequence[Union[TRTTensor, np.ndarray, torch.Tensor]]` since a single TRTTensor cannot be iterated, right?

Reply:
When you say `accumulate=True` is causing issue, I believe the duplicate indices causes issues. I faced the same in `scatter_reduce` and I believe advanced indexing would be the way to deal with it (lengthy code that would be I believe :( ). Do you have any other ideas?

Reply:
I have write a validator to handle the `accumulate=True` case. And I have created a separate [issue](https://github.com/pytorch/TensorRT/issues/2939) for implementing the converter for `accumulate=True`. It would be great to share ideas and work together on this.